extensions/keycloak-authorization/runtime/src/main/java/io/quarkus/keycloak/pep/KeycloakPolicyEnforcerAuthorizer.java

```diff
@@ -0,0 +1,191 @@
+package io.quarkus.keycloak.pep;
+
+import java.security.Permission;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.inject.Singleton;
+
+import org.keycloak.AuthorizationContext;
+import org.keycloak.adapters.KeycloakDeploymentBuilder;
+import org.keycloak.adapters.authorization.KeycloakAdapterPolicyEnforcer;
+import org.keycloak.adapters.authorization.PolicyEnforcer;
+import org.keycloak.representations.adapters.config.AdapterConfig;
+import org.keycloak.representations.adapters.config.PolicyEnforcerConfig;
+
+import io.quarkus.arc.AlternativePriority;
+import io.quarkus.oidc.OidcConfig;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.runtime.QuarkusSecurityIdentity;
+import io.quarkus.vertx.http.runtime.security.HttpAuthorizer;
+import io.quarkus.vertx.http.runtime.security.QuarkusHttpUser;
+import io.vertx.ext.web.RoutingContext;
+
+@Singleton
+@AlternativePriority(1)
+public class KeycloakPolicyEnforcerAuthorizer extends HttpAuthorizer {
+
+    private KeycloakAdapterPolicyEnforcer delegate;
+
+    @Override
+    public CompletionStage<SecurityIdentity> checkPermission(RoutingContext routingContext) {
+        VertxHttpFacade httpFacade = new VertxHttpFacade(routingContext);
+        AuthorizationContext result = delegate.authorize(httpFacade);
+
+        if (result.isGranted()) {
```

Comment:
There probably should be some abstract code based on what Stuart @stuartwdouglas did, so that the default authorizer and this one do not duplicate. 
What was the problem with only plugging in a custom `HttpSecurityPolicy` ?

Reply:
I don't think there is a duplication. The authorization process is different if you compare both solutions.

This extension just brings an existing and specific feature from Keycloak into Quarkus. It is related to centralized authorization and how to externalize authorization from applications based on a Resource-Based Access Control.